linera-service/src/cli_wrappers/local_kubernetes_net.rs

```diff
@@ -62,7 +62,29 @@ pub struct LocalKubernetesNet {
 #[cfg(any(test, feature = "test"))]
 impl SharedLocalKubernetesNetTestingConfig {
     pub fn new(network: Network, binaries: Option<Option<PathBuf>>) -> Self {
-        Self { network, binaries }
+        if binaries.is_none() {
+            // For cargo test, current binary should be in debug mode
+            let current_binary_parent =
+                current_binary_parent().expect("Fetching current binaries path should not fail");
+            // But binaries for cluster should be release mode
+            let binaries_dir = current_binary_parent
+                .parent()
+                .expect("Getting parent shuold not fail")
+                .join("release");
+            if binaries_dir.exists() {
+                // If release exists, use those binaries
```

Comment:
Can't we explicitly pass the release binary path in CI?

Reply:
The problem is that the tests will be run locally as well as in CI, which will have two different paths 😕 so if we set it explicitly for CI, we'll break it for local runs.
Now that you mention this, not even this is not great, because it means that if you usually do debug builds, and you have some random binaries built in `release` mode, it might test those instead of the debug binaries you want to be tested 😅 so let me fix that
I'll do the following: If we're in CI, set the path like we're doing now. If you're not, keep the existing default behavior (which right now just builds within the container). These are the modes, to be clear:

- `None` -> builds within the Docker container. Should work for everyone.
- `Some(None)` -> Will search for the binaries in the same directory as the current binary (for tests it'll be debug builds)
- `Some(path_dir)` -> Will search for the binaries in the path directory